### PR TITLE
GUACAMOLE-103: Document SAML authentication extension.

### PR DESCRIPTION
**Pull request by necouchman**

### CODE REVIEW

File: src/gug.xml [hunk 1]
```diff
@@ -165,6 +165,7 @@
         <xi:include href="chapters/header-auth.xml"/>
         <xi:include href="chapters/cas-auth.xml"/>
         <xi:include href="chapters/openid-auth.xml"/>
+		<xi:include href="chapters/saml-auth.xml"/>
```
Comment (mike-jumper) [inline src/gug.xml:168]: Tabs snuck in here.

File: src/chapters/saml-auth.xml [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                Guacamole as a Service Provider (SP).  Configuration of the SAML IdP is beyond
+                the scope of this document, and will vary widely based on the IdP in use.</para>
+            <variablelist>
+                <varlistentry>
+                    <term><property>saml-idp-metadata-url</property></term>
+                    <listitem>
+                        <para>The URI of the XML metadata file that from the
+                            SAML Identity Provider that contains all of the
+                            information the SAML extension needs in order to
+                            know how to authentication with the IdP. This URI
```
Comment (mike-jumper) [inline src/chapters/saml-auth.xml:72]: how to authenticate*

File: src/chapters/saml-auth.xml [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    <indexterm>
+        <primary>SAML Authentication</primary>
+    </indexterm>
+    <para>SAML is a widely implemented and used Single Sign On (SSO) provider that allows
+		applications and services to authenticate in a standard way, and brokers those
+		authentication requests to one or more back-end authentication providers.  The SAML
+		authentication extension allows Guacamole to redirect to a SAML Identity Provider (IdP)
+		for authentication and user services.  This module does not provide any capability
+		for storing or retrieving connections, and must be layered with other authentication
+		extensions that provide connection management.
```
Comment (mike-jumper) [inline src/chapters/saml-auth.xml:15]: This `<para>` is missing its closing tag, resulting in a build failure:

```
[mjumper@dev-mjumper guacamole-manual]$ make
cd src; xsltproc -o ../html/ --xinclude site.xslt gug.xml
chapters/saml-auth.xml:176: parser error : Opening and ending tag mismatch: para line 9 and chapter
</chapter>
          ^
chapters/saml-auth.xml:177: parser error : Premature end of data in tag chapter line 3

^
gug.xml:168: element include: XInclude error : could not load chapters/saml-auth.xml, and no fallback was found
make: *** [Makefile:55: html/index.html] Error 6
[mjumper@dev-mjumper guacamole-manual]$ 
```

File: src/chapters/saml-auth.xml [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+                    <term><property>saml-idp-metadata-url</property></term>
+                    <listitem>
+                        <para>The URI of the XML metadata file that from the
+                            SAML Identity Provider that contains all of the
+                            information the SAML extension needs in order to
+                            know how to authentication with the IdP. This URI
+                            can either be a remote server (e.g. https://) or a
+                            local file on the filesystem (e.g. file://). If this
+                            file is provided via this configuration property, no
+                            other configuration is required.  The metadata file is
```
Comment (mike-jumper) [inline src/chapters/saml-auth.xml:76]: > ... If this file is provided via this configuration property, no other configuration is required. ...

What about `saml-entity-id` and `saml-callback-url`? Aren't those required in all cases?

Comment (necouchman) [reply src/chapters/saml-auth.xml:76]: Sounds like they shouldn't be, so I think some additional rewording, here, and a couple of changes to the SAML `ConfigurationService` code are in order.

File: src/chapters/saml-auth.xml [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+                        <para>The URL that the IdP will use once authentication has
+                            succeeded to return to the Guacamole web application and
+                            provide the authentication details to the SAML extension.
+                            The SAML extension currently only supports callback as a
+                            POST operation to this callback URL. This property
+                            is required if either the
+                            <property>saml-idp-metadata-url</property> property
+                            is not specified, or if the provided metadata file
+                            does not contain the SAML SP Entity ID for
+                            Guacamole Client.</para>
```
Comment (mike-jumper) [inline src/chapters/saml-auth.xml:115]: I believe this property is still always required, as it drives the redirect issued from within the callback REST endpoint.

It's conceivable that it _could_ be made optional, with the redirect derived from the ACS URL, though I'm not sure how reasonable that will be to implement.

Comment (necouchman) [reply src/chapters/saml-auth.xml:115]: Ah, okay, I see.

**Pull request merged**